content/blog/nats-server-2.9.17-release.md

```diff
@@ -0,0 +1,124 @@
+
+**Relevant PRs**
+
+- [#4117](https://github.com/nats-io/nats-server/pull/4117)
+- [#4135](https://github.com/nats-io/nats-server/pull/4135)
+
+## ⏱️ Monotonic time calculations
+
+Although this improvement fixes a number of timing edge cases internal to the server, this section will serve as a Pro-Tip for anyone relying on Go's [time](https://pkg.go.dev/time) package for calculating durations using two time values.
+
+It is not uncommon to prefer UTC time values, especially in system spanning multiple time zones. However, did you know that calling the `time.Now().UTC()` will result in throwing away an internal monotonic value which differentiates two time values that are invoked within a nano-second level time frame, either in succession or concurrently? The [`time` package documentation calls this out](https://pkg.go.dev/time#hdr-Monotonic_Clocks).
+
+As a quick demonstration, the following code gets two time values in succession and then prints their values and the difference between them.
+
+```go
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+func main() {
+	t1 := time.Now()
+	t2 := time.Now()
+
+	fmt.Println("# Local (monotonic)")
+	fmt.Println(t1)
+	fmt.Println(t2)
+	fmt.Println(t2.Sub(t1))
+
+	fmt.Println("# UTC")
+	fmt.Println(t1.UTC())
+	fmt.Println(t2.UTC())
+	fmt.Println(t2.UTC().Sub(t1.UTC()))
+}
+```
+
+Below is an example output. Note the local times are differentiated by that `m=+` value shown in the output, with a calculated 83ns difference between the times. The second set of times are converted to UTC before printing and performing the time substraction. Note how the times are no longer differentiated and the difference is zero.
+
+```
+byron@nats ~ % go run time.go
+# Local (monotonic)
+2023-05-19 15:47:48.263162 -0400 EDT m=+0.000078876
+2023-05-19 15:47:48.263162 -0400 EDT m=+0.000078959
+83ns
+# UTC
+2023-05-19 19:47:48.263162 +0000 UTC
+2023-05-19 19:47:48.263162 +0000 UTC
+0s
+```
+
+UTC time values are certainly valuable for user/system reported times, however, the lesson learned here is that monotonic time values need to be preserved for accurate duration calculations.
+
+**Relevant PRs**
+
+- [#4132](https://github.com/nats-io/nats-server/pull/4132)
+- [#4154](https://github.com/nats-io/nats-server/pull/4154)
+
+## Conclusion
+
+The NATS team wants to reiterate our appreciation to all contributors in this release, including those testing and providing feedback during the release cycle. Nightly builds on the `main` branch, tracking the minor series (currently `2.9.x`), are available on Docker hub: [`synadia/nats-server:nightly-main`](https://hub.docker.com/r/synadia/nats-server/tags?page=1&name=nightly-main).
```

Comment:
nb: all our series are minor series, 2.10 is a minor series too.  Did you mean "stable series"?